src/_pytest/tmpdir.py

```diff
@@ -134,18 +135,35 @@ def get_user() -> Optional[str]:
         return None
 
 
+def pytest_configure(config) -> None:
+    """Create a TempdirFactory and attach it to the config object.
+
+    This is to comply with existing plugins which expect the handler to be
+    available at pytest_configure time, but ideally should be moved entirely
+    to the tmpdir_factory session fixture.
```

Comment:
> but ideally should be moved entirely to the tmpdir_factory session fixture.

Maybe remove this?
It cannot be moved really if plugins want to use it before fixtures are being used.

As a followup it should go into the config's store probably, and then even get documented/exposed properly.

We can merge/revert as is - but certainly the "ideally" made me think less about removing/following that advice.

Reply:
Tmpdir factory is actually the prime example for a fixture which has a component that runs on the coordinator in case of shared systems

Reply:
But it does not work / can be used when it is required before the test run already though, e.g. when used with/for startup args.
However, with xdist it could also not use `--basetemp` then, but the coordinator's component - I assume that's what you've meant?

In general though `basetemp` appears to be a useful property to provide via `config` by default, which would then either be retrieved via the tmpdir plugin, or generated via its class.  The tmpdir plugin could also instead just use the universal property from the config then - the `--basetemp` option is in `main` already, so it could handle the default there then.

Reply:
Please consider adjusting the doc so that it is clear that it is used by e.g. xdist.

Reply:
Also please squash then (assuming you are not squash-merging).